raiden/network/transport/matrix/transport.py

```diff
@@ -224,6 +227,33 @@ def notify(self) -> None:
         with self._lock:
             self._notify_event.set()
 
+    def _batch_by_address_metadata(
+        self,
+    ) -> List[Tuple[List[_MessageData], Optional[AddressMetadata]]]:
+        user_id_to_messages: Dict[Optional[UserID], List[_RetryQueue._MessageData]] = defaultdict(
+            list
+        )
+        user_id_to_metadata: Dict[UserID, AddressMetadata] = dict()
+        for message_data in self._message_queue[:]:
+            metadata = message_data.address_metadata
+            user_id: Optional[UserID] = None
+            if metadata is not None:
+                user_id = cast(Optional[UserID], metadata.get("user_id"))
+                if user_id is not None:
+                    # FIXME for N metadata with the same user-id, reduce N -> 1.
+                    #  Decide which one or how to unify:
+                    #  There can be different metadata (currently only capabilities)
+                    #  given for the same user-id.
+                    #  Currently, we don't process capabilities for sending,
+                    #  so this does not make a difference.
+                    user_id_to_metadata[user_id] = metadata
```

Comment:
I think we should treat the whole metadata object opaquely for this case. 
This is such an very edge case anyway that we shouldn't try to be too clever here.

Reply:
removed the fixme and made this look way less significant